tests/integration_tests/test_llm.py

```diff
@@ -177,3 +180,80 @@ def test_llm_zero_shot_classification(tmpdir, backend, ray_cluster_4cpu):
     assert preds["label_probabilities_positive"]
     assert preds["label_probabilities_neutral"]
     assert preds["label_probabilities_negative"]
+
+
+@pytest.mark.llm
+@pytest.mark.parametrize(
+    "backend",
+    [
+        pytest.param(LOCAL_BACKEND, id="local"),
+        pytest.param(RAY_BACKEND, id="ray"),
+    ],
+)
+def test_llm_few_shot_classification(tmpdir, backend, csv_filename, ray_cluster_4cpu):
+    input_features = [
+        text_feature(
+            output_feature=False,
+            name="body",
+            encoder={"type": "passthrough"},  # need to use the default encoder for LLMTextInputFeatureConfig
+            preprocessing={
+                "prompt": {
+                    "retrieval": {"type": "random", "k": 3},
+                    "task": (
+                        "Given the sample input, complete this sentence by replacing XXXX: The review rating is XXXX. "
+                        "Choose one value in this list: [1, 2, 3, 4, 5]."
+                    ),
+                }
+            },
+        )
+    ]
+    output_features = [
+        category_feature(
+            output_feature=True,
+            name="label",
+            preprocessing={
+                "fallback_label": "3",
+            },
+            decoder={
+                "type": "category_parser",
+                "match": {
+                    "1": {"type": "contains", "value": "1"},
+                    "2": {"type": "contains", "value": "2"},
+                    "3": {"type": "contains", "value": "3"},
+                    "4": {"type": "contains", "value": "4"},
+                    "5": {"type": "contains", "value": "5"},
+                },
+            },
+        )
+    ]
+
+    config = {
+        MODEL_TYPE: MODEL_LLM,
+        MODEL_NAME: TEST_MODEL_NAME,
+        GENERATION_CONFIG: get_generation_config(),
+        INPUT_FEATURES: input_features,
+        OUTPUT_FEATURES: output_features,
+        PREPROCESSING: {
+            "split": {"type": "fixed"},
+        },
+    }
+
+    dataset_path = generate_data(
+        input_features,
+        output_features,
+        filename=csv_filename,
+        num_examples=25,
+        nan_percent=0.1,
+        with_split=True,
+    )
+    df = pd.read_csv(dataset_path)
+    df["label"] = np.random.choice([1, 2, 3, 4, 5], size=len(df)).astype(str)  # ensure labels match the feature config
+    df.to_csv(dataset_path, index=False)
+
+    model = LudwigModel(config, backend={**backend, "cache_dir": str(tmpdir)})
+    results = model.train(dataset=dataset_path, output_directory=str(tmpdir), skip_save_processed_input=True)
+
+    # TODO: fix LLM model loading
+    # model = LudwigModel.load(os.path.join(results.output_directory, "model"), backend=backend)
+    preds, _ = model.predict(dataset=dataset_path)
+    preds = convert_preds(backend, preds)
```

Comment:
Is it worth adding asserts on `preds` to make sure we're getting back non-empty values?

Reply:
Done